src/index.js

```diff
@@ -1,34 +1,34 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { BrowserRouter } from 'react-router-dom';
+import { Router } from 'react-router-dom';
 import { ApolloProvider } from 'react-apollo';
 import { ApolloClient } from 'apollo-client';
-import { createHttpLink } from 'apollo-link-http';
+import { HttpLink } from 'apollo-link-http';
 import { InMemoryCache } from 'apollo-cache-inmemory';
+import App from 'containers/App';
+import history from './utils/history';
 
 import 'semantic-ui-css/semantic.min.css';
 import './ress.min.css';
 import './common.style';
 
-import App from 'containers/App';
-
-const uri =
-  process.env.NODE_ENV === 'production'
-    ? 'Set uri for production here'
-    : 'http://localhost:4000';
-
-const httpLink = createHttpLink({ uri });
+const uri = process.env.NODE_ENV === 'production'
+  ? 'Set uri for production here'
+  : process.env.REACT_APP_GRAPHQL_URI;
 
 const client = new ApolloClient({
-  link: httpLink,
-  cache: new InMemoryCache()
+  cache: new InMemoryCache(),
+  link: new HttpLink({
+    credentials: 'include',
+    uri,
+  }),
 });
 
 ReactDOM.render(
-  <BrowserRouter>
+  <Router history={history}>
```

Comment:
I think using `BrowserRouter` is better with `react-router v4`
`<BrowserRouter></BrowserRouter>` is same as `<Router history={browserHistory}></Router>`

https://medium.com/@ivanmontiel/changing-old-habits-in-react-router-4-8483ee5e087c

Reply:
@queq1890 The reason I use `Router` instead of `BrowserRouter` is we'll use history for custom navigate.

More detail: https://stackoverflow.com/questions/42123261/programmatically-navigate-using-react-router-v4
And don't worry about BrowserRouter and Router. It the same.